src/main/java/org/opensearch/ad/transport/AnomalyDetectorJobTransportAction.java

```diff
@@ -88,7 +88,8 @@ protected void doExecute(Task task, AnomalyDetectorJobRequest request, ActionLis
         ActionListener<AnomalyDetectorJobResponse> listener = wrapRestActionListener(actionListener, errorMessage);
 
         // By the time request reaches here, the user permissions are validated by Security plugin.
-        User user = getUserContext(client);
+        // Temporary null user for AD extension without security. Will always execute detector.
+        UserIdentity user = getNullUser();
         try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
```

Comment:
Seems like all the methods for ThreadContext were added for preserving/resetting the user from the thread context, a quick search says this can be removed.  What do you think about making that change now?

Reply:
That's actually being handled in #22.  [You can see the diff here,](https://github.com/dbwiddis/anomaly-detection/compare/remove-user...dbwiddis:anomaly-detection:remove-client) awaiting this PR to be merged. :)

I went a step further and actually removed the whole thread pool since it's not needed for anything other than the thread context, although it remains an argument in the `Plugin` constructor so would be available for multithreading applications in the future.

Reply:
Also, with the changes in the above diff, `getUserContext(client)` now returns null so that null user placeholder could be removed again... but I prefer this naming as a reminder that we need to make some new method when we've re-done the `UserIdentity` class.